src/utils/helpers.js

```diff
@@ -0,0 +1,38 @@
+// separates an array of objects into the number of desired arrays of objects
+// (data structure provided by graphData) each containing a date and value;
+// converts date strings into date objects - as required for multi-line MG graphs
+export const transformGraphData = (graphData, data) => {
+  const finalData = [];
+  for (const key of Object.keys(graphData)) {
+    for (const item of data) {
+      graphData[key].push({ date: new Date(item.date), value: item[key] });
+    }
```

Comment:
This method seems kind of strange to me. Is the graphData parameter somehow already populated with some data and we're adding more to it? It seems like we're calling this after getting some data from the server. I haven't delved into the details, but is there no way we couldn't just simplify this to a single statement (preferably inline if you're only going to use it once) which transforms the data as required? [reduce](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/reduce) is often handy if you want to build a dictionary type structure.

Reply:
I'll push a refactored version with a clearer explanatory text.